addons/binding/org.openhab.binding.squeezebox/src/main/java/org/openhab/binding/squeezebox/handler/SqueezeBoxServerHandler.java

```diff
@@ -533,6 +543,10 @@ public void updateListener(SqueezeBoxPlayerEventListener listener) {
         }
 
         private void handleStatusMessage(final String mac, String[] messageParts) {
+            String remoteTitle = "", artist = "", album = "", genre = "", year = "";
+            // default url for cover art
+            String url = "http://" + host + ":" + webport + "/music/current/cover.jpg?player=" + encode(mac);
```

Comment:
This url logic is now scattered over at least three places can it be moved to a dedicated method?

Reply:
yes I would have preferred too, but I had to deal with different cases. In most cases the default URL is sufficient, but in some cases it does not work and I need to update it (line 698)

Reply:
Please extract a method which handles this URL logic.

Reply:
I agree it would be clearer to have a single method that handled all the URL logic, including setting the default artwork URL.  Could you encapsulate all the url logic (including setting the default) in a separate method by matching on just "artwork_url%3A" below.  Or, is there an edge case that would be missed by this approach?